src/ResourceManagement/Locks/ManagementLocksImpl.cs

```diff
@@ -0,0 +1,308 @@
+        {
+            string scope = ResourceIdFromLockId(id);
+            string lockName = ResourceUtils.NameFromResourceId(id);
+            if (scope != null && lockName != null)
+            {
+                await Inner().DeleteByScopeAsync(scope, lockName);
+            }
+        }
+
+        ///GENMHASH:782853D3A86D961975AF25BD353144CE:F42FD335BA3F56DE82610403C83DB1CD
+        public async Task<IEnumerable<string>> DeleteByIdsAsync(IList<string> ids, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (ids == null || ids.Count == 0)
+            {
+                return new List<string>();
+            }
+
+            IEnumerable<Task<string>> tasks = new List<Task<string>>();
+            List<string> ids1 = new List<string>();
+            List<Task> tts = new List<Task>();
+
+            foreach (var id in ids)
+            {
+                string lockName = ResourceUtils.NameFromResourceId(id);
+                string scopeName = ResourceIdFromLockId(id);
+                if (lockName != null && scopeName != null)
+                {
+                    Task t = Inner().DeleteByScopeAsync(scopeName, lockName).ContinueWith((Task parent) =>
+                    {
+                        ids1.Add(id);
+                    });
+                    tts.Add(t);
+                }
+            }
+
+            await Task.WhenAll(tts);
+            return ids1;
+        }
+
+        ///GENMHASH:6761F083A3838E34703AD0305B873679:E05AE73A1CA90E964ED86C07093D463F
+        public async Task<IEnumerable<string>> DeleteByIdsAsync(string[] ids, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return await DeleteByIdsAsync(new List<string>(ids), cancellationToken);
+        }
+
+        ///GENMHASH:8B5750E68FCC626D3009EA1EAACB3C16:8E4631EFC6068BB521E67D3178D6E7B8
+        public void DeleteByIds(IList<string> ids)
+        {
+            Extensions.Synchronize(() => DeleteByIdsAsync(ids));
+        }
+
+        ///GENMHASH:F65FF3868FDEB2B4896429AE1A0F14F4:8E4631EFC6068BB521E67D3178D6E7B8
+        public void DeleteByIds(params string[] ids)
+        {
+            Extensions.Synchronize(() => DeleteByIdsAsync(ids));
+        }
+
+        ///GENMHASH:DE94604AE39A5722E2DA0AC4017FC2B9:EAFF99EE2BE784A51EF82B7277636D2B
+        public void DeleteByResourceGroup(string resourceGroupName, string name)
+        {
+            Extensions.Synchronize(() => DeleteByResourceGroupAsync(resourceGroupName, name));
+        }
+
+        ///GENMHASH:9D38835F71DF2C39BF88CBB588420D30:FD7FA95DB661326602615F08FEA599EC
+        public async Task DeleteByResourceGroupAsync(string resourceGroupName, string name, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await Inner().DeleteAtResourceGroupLevelAsync(resourceGroupName, name);
```

Comment:
please pass cancellationToken as well to this call.

Reply:
goo catch